### PR TITLE
docs: Update documentation on how to run client-credentails 

### PR DESCRIPTION
**Pull request by shanedell**
docs: Update documentation on how to run client-credentails

Closes https://github.com/opentdf/otdfctl/issues/430

Signed-off-by: Shane Dell <shanedell100@gmail.com>

### CODE REVIEW

File: docs/man/auth/client-credentials.md [hunk 1]
```diff
@@ -23,14 +23,20 @@ in the OS keyring for future use.
 
 ## Examples
 
+Authenticate with client credentials (id and secret provided interactively)
+
+```shell
+opentdf auth client-credentials
```
Comment (jakedoublev) [inline docs/man/auth/client-credentials.md:29]: Is there a reason for renaming the CLI here from `otdfctl` to `opentdf`?

File: go.mod [hunk 2]
```diff
@@ -79,11 +79,10 @@ require (
 	github.com/muesli/reflow v0.3.0 // indirect
 	github.com/muesli/termenv v0.15.3-0.20240618155329-98d742f6907a // indirect
 	github.com/muhlemmer/gu v0.3.1 // indirect
-	github.com/opentdf/platform/lib/ocrypto v0.1.6 // indirect
+	github.com/opentdf/platform/lib/ocrypto v0.1.7 // indirect
```
Comment (jakedoublev) [inline go.mod:82]: Can you please separate these go mod changes from the documentation update PR?

**Review by jakedoublev — COMMENTED** [review]
Thanks for opening this @shanedell.

File: docs/man/auth/client-credentials.md (hunk 1 above)
Comment (shanedell) [reply docs/man/auth/client-credentials.md]: Yeah not sure how or why I did that but I reverted them back to `otdfctl`

File: go.mod (hunk 2 above)
Comment (shanedell) [reply go.mod]: yeah didn't mean to add these will revert back

**Review by jakedoublev — APPROVED** [review]

**Comment by shanedell** [comment]
@jakedoublev would you be able to reapprove and merge? I had force pushed to sign the commit however it seems like it still might not be working. Is there something I perhaps missed?

**Pull request merged**